torch/csrc/utils/tensor_new.cpp

```diff
@@ -292,7 +294,29 @@ Tensor internal_new_from_data(
           "Expected a Storage of type ", scalar_type,
           " or an _UntypedStorage, but got ", storage_scalar_type);
       tensor = at::empty(sizes, at::initialTensorOptions().dtype(is_typed_storage ? storage_scalar_type : inferred_scalar_type).pinned_memory(pin_memory).device(storage.device()));
```

Comment:
It occurs to me that this actually is very inefficient, right?! We allocate a sizes storage, and then throw it out immediately after! If we had some sort of `new_as_strided` (which @albanD was mumbling about at https://github.com/pytorch/pytorch/pull/75994 ) we could do this all in one go, it would be faster, and you could directly implement functionalization there.

Reply:
We do have the low level function in c++: https://github.com/pytorch/pytorch/blob/1d55518198de14a011e7acd6c604afe512ee28b4/aten/src/ATen/native/native_functions.yaml#L1902
The new version could be added!